sys/mips/include/atomic.h

```diff
@@ -555,6 +537,138 @@ ATOMIC_STORE_LOAD(64)
 #define	atomic_load_64	atomic_load_acq_64
 #endif
 
+#ifdef __CHERI_PURE_CAPABILITY__
+/*
+ * In a purecap kernel we cannot use the generic sub-word implementation.
+ */
+static __inline int
```

Comment:
Maybe just use the compiler intrinsic like risc-v does? But I guess the inline asm code here exists so we might as well use it.
https://cheri-compiler-explorer.cl.cam.ac.uk/z/48xYza

Reply:
I can use the compiler intrinsics, but if I do I think we have to ditch the assembly for the variants non specific to the purecap kernel as well. It is probably for the best anyway though.